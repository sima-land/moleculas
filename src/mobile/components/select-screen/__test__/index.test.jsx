import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import SelectScreen from '../index';

describe('<SelectScreen />', () => {
  it('should render without props', () => {
    const wrapper = mount(
      <SelectScreen />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle props', () => {
    const spy = jest.fn();

    const wrapper = mount(
      <SelectScreen
        title='Районы'
        items={[
          'Верх-Исетский',
          'Чкаловский',
          'Железнодорожный',
          'Кировский',
          'Орджоникидзевский',
          'Октябрьский',
          'Ленинский',
        ]}
        getItemName={String}
        onItemClick={spy}
        isItemSelected={item => item === 'Кировский'}
      />
    );

    expect(wrapper).toMatchSnapshot();

    expect(spy).toHaveBeenCalledTimes(0);

    act(() => {
      wrapper.find('button.item').at(2).prop('onClick')();
    });

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toBeCalledWith('Железнодорожный');
  });

  it('should handle "displayItem" prop', () => {
    const wrapper = mount(
      <SelectScreen
        title='Валюта'
        items={[
          { shortName: 'RUB', name: 'Российский рубль' },
          { shortName: 'USD', name: 'Американский доллар' },
          { shortName: 'EUR', name: 'Единая европейская валюта' },
          { shortName: 'BYN', name: 'Белорусский рубль' },
          { shortName: 'UAH', name: 'Украинская гривна' },
          { shortName: 'KZT', name: 'Казахстанский тенге' },
        ]}
        isItemSelected={item => item.shortName === 'RUB'}
        displayItem={(item, { isSelected }) => (
          <div className={isSelected ? 'item is-selected' : 'item'}>
            {item.name} ({item.shortName})
          </div>
        )}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle miss "getItemName" prop', () => {
    const wrapper = mount(
      <SelectScreen
        title='Валюта'
        items={[
          { name: 'Российский рубль' },
          { name: 'Американский доллар' },
          { name: 'Единая европейская валюта' },
          { name: 'Белорусский рубль' },
          { name: 'Украинская гривна' },
          { name: 'Казахстанский тенге' },
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
