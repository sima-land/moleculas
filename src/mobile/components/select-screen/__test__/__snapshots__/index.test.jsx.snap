// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectScreen /> should handle "displayItem" prop 1`] = `
<SelectScreen
  displayItem={[Function]}
  isItemSelected={[Function]}
  items={
    [
      {
        "name": "Российский рубль",
        "shortName": "RUB",
      },
      {
        "name": "Американский доллар",
        "shortName": "USD",
      },
      {
        "name": "Единая европейская валюта",
        "shortName": "EUR",
      },
      {
        "name": "Белорусский рубль",
        "shortName": "BYN",
      },
      {
        "name": "Украинская гривна",
        "shortName": "UAH",
      },
      {
        "name": "Казахстанский тенге",
        "shortName": "KZT",
      },
    ]
  }
  title="Валюта"
>
  <Screen>
    <div
      className="root"
      data-testid="screen"
      style={
        {
          "zIndex": 300,
        }
      }
    >
      <HeaderSlot
        key=".0"
        title="Валюта"
      >
        <NavBar
          buttons={
            {
              "end": undefined,
              "start": undefined,
            }
          }
          title="Валюта"
        >
          <div
            className="nav-bar"
            style={
              {
                "height": "56px",
              }
            }
          >
            <div
              className="main"
            >
              <div
                className="title truncate"
              >
                Валюта
              </div>
            </div>
          </div>
        </NavBar>
      </HeaderSlot>
      <ForwardRef(BodySlot)
        key=".1"
      >
        <div
          className="body"
          data-bsl-ignore={true}
          data-testid="screen:body"
        >
          <SelectScreenLayout
            style={
              {
                "padding": "16px 0",
              }
            }
          >
            <MobileLayout
              disabledOn={
                [
                  "mxs",
                  "ms",
                ]
              }
              style={
                {
                  "padding": "16px 0",
                }
              }
            >
              <div
                className="layout mobile disabled-mxs disabled-ms"
                style={
                  {
                    "padding": "16px 0",
                  }
                }
              >
                <div
                  className="container"
                >
                  <SelectScreenOption
                    key=".$0"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <div
                        className="item is-selected"
                      >
                        Российский рубль
                         (
                        RUB
                        )
                      </div>
                      <ForwardRef(CheckSVG)
                        fill="#212121"
                      >
                        <svg
                          fill="#212121"
                          height={24}
                          viewBox="0 0 24 24"
                          width={24}
                        >
                          <path
                            clipRule="evenodd"
                            d="M21.678 5.265a1 1 0 01.057 1.413l-12 13a1 1 0 01-1.36.103l-5-4a1 1 0 011.25-1.562l4.272 3.418L20.265 5.322a1 1 0 011.413-.057z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </ForwardRef(CheckSVG)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$1"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <div
                        className="item"
                      >
                        Американский доллар
                         (
                        USD
                        )
                      </div>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$2"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <div
                        className="item"
                      >
                        Единая европейская валюта
                         (
                        EUR
                        )
                      </div>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$3"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <div
                        className="item"
                      >
                        Белорусский рубль
                         (
                        BYN
                        )
                      </div>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$4"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <div
                        className="item"
                      >
                        Украинская гривна
                         (
                        UAH
                        )
                      </div>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$5"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <div
                        className="item"
                      >
                        Казахстанский тенге
                         (
                        KZT
                        )
                      </div>
                    </button>
                  </SelectScreenOption>
                </div>
              </div>
            </MobileLayout>
          </SelectScreenLayout>
        </div>
      </ForwardRef(BodySlot)>
    </div>
  </Screen>
</SelectScreen>
`;

exports[`<SelectScreen /> should handle miss "getItemName" prop 1`] = `
<SelectScreen
  items={
    [
      {
        "name": "Российский рубль",
      },
      {
        "name": "Американский доллар",
      },
      {
        "name": "Единая европейская валюта",
      },
      {
        "name": "Белорусский рубль",
      },
      {
        "name": "Украинская гривна",
      },
      {
        "name": "Казахстанский тенге",
      },
    ]
  }
  title="Валюта"
>
  <Screen>
    <div
      className="root"
      data-testid="screen"
      style={
        {
          "zIndex": 300,
        }
      }
    >
      <HeaderSlot
        key=".0"
        title="Валюта"
      >
        <NavBar
          buttons={
            {
              "end": undefined,
              "start": undefined,
            }
          }
          title="Валюта"
        >
          <div
            className="nav-bar"
            style={
              {
                "height": "56px",
              }
            }
          >
            <div
              className="main"
            >
              <div
                className="title truncate"
              >
                Валюта
              </div>
            </div>
          </div>
        </NavBar>
      </HeaderSlot>
      <ForwardRef(BodySlot)
        key=".1"
      >
        <div
          className="body"
          data-bsl-ignore={true}
          data-testid="screen:body"
        >
          <SelectScreenLayout
            style={
              {
                "padding": "16px 0",
              }
            }
          >
            <MobileLayout
              disabledOn={
                [
                  "mxs",
                  "ms",
                ]
              }
              style={
                {
                  "padding": "16px 0",
                }
              }
            >
              <div
                className="layout mobile disabled-mxs disabled-ms"
                style={
                  {
                    "padding": "16px 0",
                  }
                }
              >
                <div
                  className="container"
                >
                  <SelectScreenOption
                    key=".$0"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Российский рубль
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$1"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Американский доллар
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$2"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Единая европейская валюта
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$3"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Белорусский рубль
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$4"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Украинская гривна
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$5"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Казахстанский тенге
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                </div>
              </div>
            </MobileLayout>
          </SelectScreenLayout>
        </div>
      </ForwardRef(BodySlot)>
    </div>
  </Screen>
</SelectScreen>
`;

exports[`<SelectScreen /> should handle props 1`] = `
<SelectScreen
  getItemName={[Function]}
  isItemSelected={[Function]}
  items={
    [
      "Верх-Исетский",
      "Чкаловский",
      "Железнодорожный",
      "Кировский",
      "Орджоникидзевский",
      "Октябрьский",
      "Ленинский",
    ]
  }
  onItemClick={[MockFunction]}
  title="Районы"
>
  <Screen>
    <div
      className="root"
      data-testid="screen"
      style={
        {
          "zIndex": 300,
        }
      }
    >
      <HeaderSlot
        key=".0"
        title="Районы"
      >
        <NavBar
          buttons={
            {
              "end": undefined,
              "start": undefined,
            }
          }
          title="Районы"
        >
          <div
            className="nav-bar"
            style={
              {
                "height": "56px",
              }
            }
          >
            <div
              className="main"
            >
              <div
                className="title truncate"
              >
                Районы
              </div>
            </div>
          </div>
        </NavBar>
      </HeaderSlot>
      <ForwardRef(BodySlot)
        key=".1"
      >
        <div
          className="body"
          data-bsl-ignore={true}
          data-testid="screen:body"
        >
          <SelectScreenLayout
            style={
              {
                "padding": "16px 0",
              }
            }
          >
            <MobileLayout
              disabledOn={
                [
                  "mxs",
                  "ms",
                ]
              }
              style={
                {
                  "padding": "16px 0",
                }
              }
            >
              <div
                className="layout mobile disabled-mxs disabled-ms"
                style={
                  {
                    "padding": "16px 0",
                  }
                }
              >
                <div
                  className="container"
                >
                  <SelectScreenOption
                    key=".$0"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Верх-Исетский
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$1"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Чкаловский
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$2"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Железнодорожный
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$3"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Кировский
                        </span>
                      </ForwardRef(Text)>
                      <ForwardRef(CheckSVG)
                        fill="#212121"
                      >
                        <svg
                          fill="#212121"
                          height={24}
                          viewBox="0 0 24 24"
                          width={24}
                        >
                          <path
                            clipRule="evenodd"
                            d="M21.678 5.265a1 1 0 01.057 1.413l-12 13a1 1 0 01-1.36.103l-5-4a1 1 0 011.25-1.562l4.272 3.418L20.265 5.322a1 1 0 011.413-.057z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </ForwardRef(CheckSVG)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$4"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Орджоникидзевский
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$5"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Октябрьский
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                  <SelectScreenOption
                    key=".$6"
                    onClick={[Function]}
                  >
                    <button
                      className="item size-m"
                      data-testid="select-screen:option"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(Text)
                        color="basic-gray87"
                        lineHeight={24}
                        size={16}
                      >
                        <span
                          className="size-16 line-height-24 color__basic-gray87"
                          style={{}}
                        >
                          Ленинский
                        </span>
                      </ForwardRef(Text)>
                    </button>
                  </SelectScreenOption>
                </div>
              </div>
            </MobileLayout>
          </SelectScreenLayout>
        </div>
      </ForwardRef(BodySlot)>
    </div>
  </Screen>
</SelectScreen>
`;

exports[`<SelectScreen /> should render without props 1`] = `
<SelectScreen>
  <Screen>
    <div
      className="root"
      data-testid="screen"
      style={
        {
          "zIndex": 300,
        }
      }
    >
      <HeaderSlot
        key=".0"
      >
        <NavBar
          buttons={
            {
              "end": undefined,
              "start": undefined,
            }
          }
        >
          <div
            className="nav-bar"
            style={
              {
                "height": "56px",
              }
            }
          >
            <div
              className="main"
            />
          </div>
        </NavBar>
      </HeaderSlot>
      <ForwardRef(BodySlot)
        key=".1"
      >
        <div
          className="body"
          data-bsl-ignore={true}
          data-testid="screen:body"
        >
          <SelectScreenLayout
            style={
              {
                "padding": "16px 0",
              }
            }
          >
            <MobileLayout
              disabledOn={
                [
                  "mxs",
                  "ms",
                ]
              }
              style={
                {
                  "padding": "16px 0",
                }
              }
            >
              <div
                className="layout mobile disabled-mxs disabled-ms"
                style={
                  {
                    "padding": "16px 0",
                  }
                }
              >
                <div
                  className="container"
                />
              </div>
            </MobileLayout>
          </SelectScreenLayout>
        </div>
      </ForwardRef(BodySlot)>
    </div>
  </Screen>
</SelectScreen>
`;

exports[`SelectScreenOption should handle size prop 1`] = `
<div>
  <button
    class="item size-xl"
    data-testid="select-screen:option"
    type="button"
  >
    Test option
  </button>
</div>
`;

exports[`select screen parts should works properly 1`] = `
<div>
  <div
    class="root"
    data-testid="screen"
    style="z-index: 300;"
  >
    <div
      class="body"
      data-bsl-ignore="true"
      data-testid="screen:body"
    >
      <div
        class="layout mobile disabled-mxs disabled-ms"
        style="padding: 16px 0px;"
      >
        <div
          class="container"
        >
          <button
            class="item size-m"
            data-testid="select-screen:option"
            type="button"
          >
            Foo
          </button>
          <button
            class="item size-m"
            data-testid="select-screen:option"
            type="button"
          >
            Bar
            <svg
              class="icon"
              height="24"
              role="presentation"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                clip-rule="evenodd"
                d="M21.678 5.265a1 1 0 01.057 1.413l-12 13a1 1 0 01-1.36.103l-5-4a1 1 0 011.25-1.562l4.272 3.418L20.265 5.322a1 1 0 011.413-.057z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <button
            class="item size-m"
            data-testid="select-screen:option"
            type="button"
          >
            Baz
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
